pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -2893,3 +2921,220 @@ func TestGetIntOrPercentValue(t *testing.T) {
 func IntPtr(i int) *int {
 	return &i
 }
+
+func assertEvents(t *testing.T, testCase string, expectedEvents []string, realEvents chan string) {
+	if len(expectedEvents) != len(realEvents) {
+		t.Errorf(
+			"Test case: %s. Number of expected events (%v) differs from number of real events (%v)",
+			testCase,
+			len(expectedEvents),
+			len(realEvents),
+		)
+	} else {
+		for _, eventType := range expectedEvents {
+			select {
+			case event := <-realEvents:
+				if !strings.Contains(event, fmt.Sprintf(" %s ", eventType)) {
+					t.Errorf("Test case: %s. Expected %v event, got: %v", testCase, eventType, event)
+				}
+			default:
+				t.Errorf("Test case: %s. Expected %v event, but no event occured", testCase, eventType)
+			}
+		}
+	}
+}
+
+// newFakeReconciler returns a new reconcile.Reconciler with a fake client
+func newFakeReconciler(initObjects ...runtime.Object) *ReconcileMachineHealthCheck {
+	return newFakeReconcilerWithCustomRecorder(nil, initObjects...)
+}
+
+func newFakeReconcilerWithCustomRecorder(recorder record.EventRecorder, initObjects ...runtime.Object) *ReconcileMachineHealthCheck {
+	fakeClient := fake.NewClientBuilder().WithRuntimeObjects(initObjects...).Build()
+	return &ReconcileMachineHealthCheck{
+		client:    &clientWrap{Client: fakeClient},
+		scheme:    scheme.Scheme,
+		namespace: namespace,
+		recorder:  recorder,
+	}
+}
+
+func assertBaseReconcile(t *testing.T, tc testCase, ctx context.Context, r *ReconcileMachineHealthCheck) {
+	recorder := r.recorder.(*record.FakeRecorder)
+
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: tc.mhc.GetNamespace(),
+			Name:      tc.mhc.GetName(),
+		},
+	}
+	result, err := r.Reconcile(ctx, request)
+	if &result == nil {
+		t.Errorf("Test case: %s. Expected: non nil result error, got: nil", tc.node.Name)
+	}
+	assertEvents(t, tc.name, tc.expectedEvents, recorder.Events)
+	if tc.expected.error != (err != nil) {
+		var errorExpectation string
+		if !tc.expected.error {
+			errorExpectation = "no"
+		}
+		t.Errorf("Test case: %s. Expected: %s error, got: %v", tc.node.Name, errorExpectation, err)
+	}
+
+	if result != tc.expected.result {
+		if tc.expected.result.Requeue {
+			before := tc.expected.result.RequeueAfter
+			after := tc.expected.result.RequeueAfter + time.Second
+			if after < result.RequeueAfter || before > result.RequeueAfter {
+				t.Errorf("Test case: %s. Expected RequeueAfter between: %v and %v, got: %v", tc.node.Name, before, after, result)
+			}
+		} else {
+			t.Errorf("Test case: %s. Expected: %v, got: %v", tc.node.Name, tc.expected.result, result)
+		}
+	}
+	g := NewWithT(t)
+	if tc.expectedStatus != nil {
+		mhc := &mapiv1beta1.MachineHealthCheck{}
+		g.Expect(r.client.Get(ctx, request.NamespacedName, mhc)).To(Succeed())
+		g.Expect(tc.expectedStatus).To(MatchMachineHealthCheckStatus(&mhc.Status))
+	}
+}
```

Comment:
This all seems to have been moved without real need, cosmetic changes again? Can we move it back or at least keep cosmetic changes to a separate commit, currently this makes it hard to review with so many changes in this one commit

Reply:
You are right. my apologies.
In my the next PR (the short-circuit back off) I've already concentrated all the tidy changes into one  commit.